### PR TITLE
Implemented login, logout and register.

### PR DESCRIPTION
**Pull request by AlbertoCasasOrtiz**
I have implemented frontend pages for login, logout and register, and views in the backend to do this.

I have a question. I have implemented login functionality based in tokens. When a user logs in, the backend creates a token (and stores it in the database), and sends it to the frontend. This way, a session is linked to a specific token per user and the frontend knows if an user is authenticated or not based on the existence of the token. I initially created a state called isLoggedin, that is set to true when you log in, and to false when you log out, but I feel it is useless now. Should we just create a function called isLoggedin, that checks if the token exists?

**Note**: It is necessary to do makemigrations/migrate, since I used django tokens for the login functionality.
**Note 2**: This is a simple implementation that works. We need to implement timeout dates for tokens as a security measure (so users are not logged in forever), a refresh token mechanism, and ways of checking if the token exists or not in the backend. We could create a function that takes as input the target URL, the parameters, and takes care of everything.
**Note 3**: This function from *Note 2* would also take care of detecting if the token is consistent or not with what we have in the backend, since any axios call should return 401 or 403 if the token does not exist, or the access to a resource is forbidden.

### CODE REVIEW

**Comment by aymanhab** [comment]
Thanks @AlbertoCasasOrtiz I agree the state is unnecessary if we can do without would be better. Let me know if you remove it, and fix lint errors and then I'll merge and build on it for my other downstream PRs. Thank you

File: src/components/pages/LogoutPage.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    const { t } = useTranslation();
+    const navigate = useNavigate();
+
+    const [errorMessage, setErrorMessage] = useState<string | null>('');
+
+    const handleLogout = async () => {
+        try {
+            if (isLoggedin) {
+                let response = await axios.post('http://localhost:8000/logout/', null, {
+                    headers: {
```
Comment (aymanhab) [inline src/components/pages/LogoutPage.tsx:24]: Would be good to move all the localhost:8000 into one place in settings

File: src/components/pages/LogoutPage.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                      Authorization: `Token ${localStorage.getItem('token')}`,
+                    },
+                });
+                if (response.status === 200) {
```
Comment (aymanhab) [inline src/components/pages/LogoutPage.tsx:28]: if this is async, and we're awaiting should this block go into .then?

Comment (AlbertoCasasOrtiz) [reply src/components/pages/LogoutPage.tsx]: You are right, I just moved it inside of .then. I checked in the other files and there was the same bug in the Register page.

File: src/components/pages/LogoutPage.tsx (hunk 1 above)
Comment (AlbertoCasasOrtiz) [reply src/components/pages/LogoutPage.tsx]: I agree, we should not be manually edit this everywhere everytime it changes.

I was not sure what you meant by settings (I know we have a settings.py file in the backend, but I am not sure about if we have similar in the frontend), so I created a "urlHelpers" file with functions to retrieve automatically protocol, hostname and everything related to the URL. Assuming the backend is in the same host, with a different port, I created a function with the port hardcoded, but that fills everything else automaticallyu (maybe we can convert the port and hostname into an environment variable and we could set an environment or config file?).

**Review by aymanhab — APPROVED** [review]
LGTM, thanks @AlbertoCasasOrtiz

**Pull request merged**